diff --git a/include/wx/eventfilter.h b/include/wx/eventfilter.h
index f105b0c0b1..9e51ccd9a5 100644
--- a/include/wx/eventfilter.h
+++ b/include/wx/eventfilter.h
@@ -19,7 +19,7 @@ class WXDLLIMPEXP_FWD_BASE wxEvtHandler;
 // wxEventFilter is used with wxEvtHandler::AddFilter() and ProcessEvent().
 // ----------------------------------------------------------------------------
 
-class wxEventFilter
+class WXDLLIMPEXP_BASE wxEventFilter
 {
 public:
     // Possible return values for FilterEvent().
@@ -38,15 +38,9 @@ public:
         Event_Processed = 1
     };
 
-    wxEventFilter()
-    {
-        m_next = nullptr;
-    }
+    wxEventFilter();
 
-    virtual ~wxEventFilter()
-    {
-        wxASSERT_MSG( !m_next, "Forgot to call wxEvtHandler::RemoveFilter()?" );
-    }
+    virtual ~wxEventFilter();
 
     // This method allows to filter all the events processed by the program, so
     // you should try to return quickly from it to avoid slowing down the
diff --git a/src/common/event.cpp b/src/common/event.cpp
index 5bff35db86..4d7ef7987c 100644
--- a/src/common/event.cpp
+++ b/src/common/event.cpp
@@ -1184,6 +1184,20 @@ void wxEventHashTable::GrowEventTypeTable()
     delete[] oldEventTypeTable;
 }
 
+// ----------------------------------------------------------------------------
+// wxEventFilter
+// ----------------------------------------------------------------------------
+
+wxEventFilter::wxEventFilter()
+{
+    m_next = nullptr;
+}
+
+wxEventFilter::~wxEventFilter()
+{
+    wxASSERT_MSG( !m_next, "Forgot to call wxEvtHandler::RemoveFilter()?" );
+}
+
 // ----------------------------------------------------------------------------
 // wxEvtHandler
 // ----------------------------------------------------------------------------
